src/components/TextInput.tsx

```diff
@@ -0,0 +1,99 @@
+import { TextIcon, icons } from "./TextIcon";
+
+type TextInputProps = {
+    value: string;
+    onChange: (value: string) => void;
+    placeholderText?: string;
+    isPassword?: boolean;
+    isDisabled?: boolean;
+    prefixIcon?: keyof typeof icons;
+    suffixIcon?: keyof typeof icons;
+    validationState?: "error" | "success" | null;
+    autoIcons?: boolean;
+}
+
+/** 
+ * Icon options include "lock", "error", "euro", "at" (@).
+ * 
+ * Obvious icons (e.g. for password input) are added automatically.
+ * Send autoIcons = false to switch this off.
+ * 
+ * Features still to be added:
+ * 
+ * - Focus: border color change, shadow effect.
+ * - Error: shake animation.
+ * 
+ */
+export const TextInput = ({
+    value,
+    onChange,
+    placeholderText,
+    isPassword,
+    isDisabled,
+    prefixIcon,
+    suffixIcon,
+    validationState = null,
+    autoIcons = true
+}: TextInputProps) => {
+
+    // Use placeholder text if provided, or use something appropriate
+    const placeholderTextFinal = placeholderText
+        ? placeholderText
+        : isPassword
+            ? "Enter password..."
```

Comment:
why is password being used here as the "something appropriate"?

Reply:
This makes sense; if it's a password input, the placeholder text defaults to "Enter password...", otherwise it's a generic "enter text"

We could break this code out slightly more by making e.g.

`const placeholderTextDefault = isPassword ? "Enter password..." : "Enter text..." `

(extendable with more defaults if we had more types/cases for what the placeholder text should be)

then 

`const placeholderTextDisplay = placeholderText || placeholderTextDefault`